src/network/networkd-dhcp4.c

```diff
@@ -1286,6 +1286,29 @@ static int dhcp4_set_client_identifier(Link *link) {
         return 0;
 }
 
+static int dhcp4_set_request_address(Link *link) {
+        Address *a;
+
+        assert(link);
+        assert(link->network);
+        assert(link->dhcp_client);
+
+        if (!FLAGS_SET(link->network->keep_configuration, KEEP_CONFIGURATION_DHCP))
+                return 0;
+
+        SET_FOREACH(a, link->addresses_foreign) {
+                if (a->family != AF_INET)
+                        continue;
+                if (link_address_is_dynamic(link, a))
+                        break;
```

Comment:
Should we maybe log here that this address is picked? In case we ever have more than one dynamic address, this could be quite confusing why `sd_dhcp_client_set_request_address` was called?

Reply:
#17798.